app/v2/template/get_template.py

```diff
@@ -0,0 +1,31 @@
+import uuid
+
+from flask import jsonify, request
+from werkzeug.exceptions import abort
+
+from app import api_user
+from app.dao import templates_dao
+from app.schema_validation import validate
+from app.v2.template import template_blueprint
+from app.v2.template.template_schemas import get_template_by_id_request
+
+
+@template_blueprint.route("/<template_id>", methods=['GET'])
+@template_blueprint.route("/<template_id>/version/<version>", methods=['GET'])
+def get_template_by_id(template_id, version=None):
+    try:
+        casted_id = uuid.UUID(template_id)
+
```

Comment:
We can probably remove this cast/check as the schema will attempt to validate this anyway?